service/client/temporal/client.go

```diff
@@ -337,10 +337,13 @@ func (t *temporalClient) DescribeWorkflowExecution(
 
 	memo, err := t.getMemoAndDecryptIfNeeded(resp.GetWorkflowExecutionInfo().GetMemo())
 
+	wfType := resp.GetWorkflowExecutionInfo().GetType().GetName()
```

Comment:
This is  not the type that we should return. This is Temporal workflow type. We need to return IwfWorkflowType which stored as a search attribute. The describeWFExecution API call return it in the SAs list at line 333.